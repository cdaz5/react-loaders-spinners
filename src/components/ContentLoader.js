import React from 'react';
import styled, { keyframes } from 'styled-components';
import { number, bool } from 'prop-types';

const fadeLeftRight = props =>
  keyframes`
    from {
      background-position: -${props.width}px 0;
    }
    to {
      background-position: ${props.width}px 0;
    }
`;

const Content = styled.div`
  display: ${props => (props.loading ? 'flex' : 'none')};
  flex-direction: column;
  background: linear-gradient(to right, #eeeeee 8%, #e2e2e2 18%, #eeeeee 33%);
  animation: ${props => `${fadeLeftRight(props)} 2s linear infinite`};
  width: 100%;

    div {
      border: 8px solid #fff;
      height: ${props => (props.lineHeight ? `${props.lineHeight}px` : '20px')};
    }
}
`;

const divFactory = (num) => {
  const divs = [];
  for (let i = num; i > 0; i--) {
    divs.push(<div key={i} />);
  }
  return divs;
};

const ContentLoader = ({ rows, ...props }) => <Content {...props}>{divFactory(rows)}</Content>;

ContentLoader.propTypes = {
  width: number.isRequired,
  loading: bool.isRequired,
  rows: number.isRequired,
  lineHeight: number
};

ContentLoader.defaultProps = {
  width: 200,
  rows: 4,
  loading: true,
  lineHeight: 20
};

export default ContentLoader;
