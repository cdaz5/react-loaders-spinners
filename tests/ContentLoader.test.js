import React from 'react';
import renderer from 'react-test-renderer';

import ContentLoader from '../src/components/ContentLoader';

describe('ContentLoader', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<ContentLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
