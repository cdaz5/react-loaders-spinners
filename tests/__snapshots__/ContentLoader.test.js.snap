// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoader matches snapshot 1`] = `
<div
  className="sc-bdVaJa huGOCs"
  width={200}
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
